core/src/main/java/org/eclipse/hono/util/JsonHelper.java

```diff
@@ -55,4 +58,100 @@ private JsonHelper() {
             return defaultValue;
         }
     }
+
+    /**
+     * Gets a value from the given JSON Array.
+     *
+     * @param jsonArray The JSON Array to get the value from.
+     * @param pos The position to return the value from.
+     * @param defaultValue A default value to return if the pos is larger than the array
+     *        or it the value is of an unexpected type.
+     * @param clazz The target type.
+     * @param <T> The type of the value.
+     * @return The value or the given default value if pos is larger than the array or if the value is of an unexpected type.
+     * @throws NullPointerException if any of the parameters except defaultValue is {@code null}.
+     */
+    public static <T> T getValue(final JsonArray jsonArray, final int pos, final Class<T> clazz,
+                                 final T defaultValue) {
+        Objects.requireNonNull(pos);
+        Objects.requireNonNull(jsonArray);
+
+        if (pos > jsonArray.size()) {
+            LOG.debug("index {} larger than array. Actual size is {}", pos, jsonArray.size());
+            return defaultValue;
+        }
+
+        final Object value = jsonArray.getValue(pos);
+        if (value == null) {
+            return defaultValue;
+        }
+        try {
+            return clazz.cast(value);
+        } catch (final ClassCastException e) {
+            LOG.debug("unexpected value type for position [{}]: {}; expected: {}", pos, value.getClass().getSimpleName(),
+                    clazz.getSimpleName());
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Return the value designated from a JsonPath in the given JsonObject.
+     *
+     * @param jsonObject The JSON object to get the value from.
+     * @param path The JsonPath representation of the value to extract.
```

Comment:
FMPOV we need a definition of the path syntax. Is there an existing definition somewhere online which we can refer to?